Add Custom Config File Support

StoneTapeStudios opened the pull request:
# Add Custom Config File Support

Introduces a new `--config-file` flag to the Supabase CLI, enabling users to specify alternative configuration files. This enhancement allows running multiple parallel local development environments with different configurations (https://github.com/supabase/cli/issues/2139).

## Example Usage

```bash
supabase start --config-file=supabase/config.test.toml
```

## Benefits
- Supports multiple local database instances with different configurations
- Simplifies management of different environment setups (e.g., testing, staging)
- Maintains backwards compatibility with default config path

StoneTapeStudios committed:
init commit, added configFile to global flags

StoneTapeStudios committed:
fixed adding flag

StoneTapeStudios committed:
test(start): add test for custom config path

StoneTapeStudios committed:
revert rename of local var to flags and import flags under alias

StoneTapeStudios committed:
formatting

StoneTapeStudios committed:
Merge pull request #1 from StoneTapeStudios/feature/custom-config-flag

Feature/custom config flag

StoneTapeStudios committed:
formatting changes

StoneTapeStudios committed:
Merge branch 'develop' into develop

sweatybridge reviewed (commented):
Generally agree to add this functionality. However, the current implementation doesn't handle [changing workdir](https://github.com/supabase/cli/blob/develop/cmd/root.go#L94) based on the given config path.

StoneTapeStudios committed:
feat(cli): handle workdir based on config file path

Adds support for setting the working directory based on the provided config file path:
- Handles both absolute and relative config paths
- Normalizes paths across platforms
- Maintains Windows compatibility by handling drive letters
- Preserves existing behavior when no config file is specified

This addresses the feedback about workdir handling in the CLI by ensuring the
working directory is properly set based on the config file location.

Tests added:
- Relative config paths
- Absolute config paths
- Non-existent config paths
- Malformed config files

StoneTapeStudios committed:
Merge pull request #2 from StoneTapeStudios/fix/handle-changing-workdir

feat(cli): handle workdir based on config file path

StoneTapeStudios commented:
> Generally agree to add this functionality. However, the current implementation doesn't handle [changing workdir](https://github.com/supabase/cli/blob/develop/cmd/root.go#L94) based on the given config path.

Thanks for the feedback! I think I have addressed the issues with my latest changes.
